### DISCUSSION TITLE
How to use animation in render?

### DISCUSSION THREAD

**Discussion by j4587698** [AvaloniaUI/Avalonia#8794] [Q&A]
My English is not good, so most of the content is machine translation.
I want to create a transition animation in the render method.
Do you have any good ideas?
thanks a lot.

**Comment by timunie** [comment]
I think the render method is to render a single frame. So imo not a good place to run an animation.

**Reply by timunie** [reply]
https://docs.avaloniaui.net/docs/animations

**Comment by j4587698** [comment]
@timunie 
Thank you for your reply.
I can use `Dispatcher.UIThread.Post(InvalidateMeasure, DispatcherPriority.Background);` to refresh render.
But is there any way I can get the position of the animation directly? Such as the position of the current frame. Instead of using a timer to determine the position?

**Reply by timunie** [reply]
I've never done that and tbh.

What exactly is your goal? Maybe there is an easier way

**Reply by j4587698** [reply]
@timunie I want a two-layer structure, similar to book turning. I hope there will be a page turning animation when clicking. Maybe it's better for me to use the panel directly and put multiple components in it?

**Reply by timunie** [reply]
Hve a look at this Control: 
https://docs.avaloniaui.net/docs/controls/transitioningcontentcontrol

You can also write your own Transition: 
https://docs.avaloniaui.net/docs/animations/pagetransitions#example 

Maybe you can have two of these Control which are each a page?

**Reply by j4587698** [reply]
tks, I'll try.It looks much simpler
